crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -362,15 +362,55 @@ def nodes
 
       protected
 
+      # If there's separate network cluster, we have touch it before we start upgrade of other
+      # nodes, specificaly we need to evacuate the network routers from the first network node.
+      def prepare_network_node(network_node)
+        return if network_node.upgraded?
+
+        evacuate_network_node(network_node, network_node)
```

Comment:
The only other point I can think of is that *controller (API) nodes* have `.openrc`. For network node i have to find some way of bringing it.